src/compiler/crystal/macros/methods.cr

```diff
@@ -758,6 +758,60 @@ module Crystal
           end
           BoolLiteral.new(@value.includes?(piece))
         end
+      when "scan"
+        interpret_check_args do |arg|
+          unless arg.is_a?(RegexLiteral)
+            raise "StringLiteral#scan expects a regex, not #{arg.class_desc}"
+          end
+
+          regex_value = arg.value
+          if regex_value.is_a?(StringLiteral)
+            regex = Regex.new(regex_value.value, arg.options)
+          else
+            raise "regex interpolations not yet allowed in macros"
```

Comment:
```suggestion
            raise "Regex interpolations not yet allowed in macros"
```

Reply:
This code is duplicate verbatim from existing locations in this file. We shouldn't require changing it in this single instance.
Instead we should probably refactor this as a follow-up.